e2e/testcases/private_cert_secret_test.go

```diff
@@ -271,3 +277,86 @@ func TestPrivateCertSecretV1Beta1(t *testing.T) {
 		nt.T.Fatal(err)
 	}
 }
+
+func TestPrivateCertSecretWatch(t *testing.T) {
+	nt := nomostest.New(t, ntopts.SkipMonoRepo, ntopts.SkipNonLocalGitProvider,
+		ntopts.NamespaceRepo(backendNamespace, configsync.RepoSyncName))
+	nt.WaitForRepoSyncs()
+	key := "GIT_SSL_CAINFO"
+	privateCertSecret := "git-cert-pub"
+	privateCertPath := "/etc/private-cert/cert"
+	var err error
+
+	// verify the deployment doesn't have the key yet
+	err = nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, &appsv1.Deployment{}, nomostest.DeploymentMissingEnvVar(reconcilermanager.GitSync, key))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	err = nt.Validate(core.NsReconcilerName(backendNamespace, configsync.RepoSyncName), v1.NSConfigManagementSystem, &appsv1.Deployment{}, nomostest.DeploymentMissingEnvVar(reconcilermanager.GitSync, key))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nn := nomostest.RepoSyncNN(backendNamespace, configsync.RepoSyncName)
+	repoSyncBackend := nomostest.RepoSyncObjectV1Beta1FromNonRootRepo(nt, nn)
+	reconcilerName := core.NsReconcilerName(backendNamespace, configsync.RepoSyncName)
+
+	// Set RepoSync to use HTTPS with privateCertSecret
+	repoSyncHTTPS := "https://test-git-server.config-management-system-test/git/backend/repo-sync"
+	repoSyncBackend.Spec.Git.Repo = repoSyncHTTPS
+	repoSyncBackend.Spec.Git.Auth = "none"
+	repoSyncBackend.Spec.Git.SecretRef.Name = ""
+	repoSyncBackend.Spec.Git.PrivateCertSecret.Name = privateCertSecret
+	nt.RootRepos[configsync.RootSyncName].Add(nomostest.StructuredNSPath(backendNamespace, configsync.RepoSyncName), repoSyncBackend)
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Update backend RepoSync use HTTPS with privateCertSecret")
+	// RepoSync should fail without privateCertSecret
```

Comment:
RepoSync should NOT fail as privateCertSecret is set.

Reply:
Done